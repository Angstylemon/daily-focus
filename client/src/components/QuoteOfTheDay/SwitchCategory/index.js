import PropTypes from "prop-types";
import { Fab } from "@material-ui/core";
import ArrowBackIcon from "@material-ui/icons/ArrowBack";
import RadioButtonCheckedIcon from "@material-ui/icons/RadioButtonChecked";
import RadioButtonUncheckedIcon from "@material-ui/icons/RadioButtonUnchecked";
import style from "./style.module.scss";

function SwitchCategory({ cancelClicked, updateCategory, currentCategory }) {
    return (
        <div className={style.container}>
            <div className={style.listHolder}>
                <div
                    className={style.categoryOption}
                    onClick={() => updateCategory("Taylor Swift")}
                >
                    Taylor Swift
                    {currentCategory === "Taylor Swift" ? (
                        <RadioButtonCheckedIcon />
                    ) : (
                        <RadioButtonUncheckedIcon />
                    )}
                </div>
                <div className={style.categoryOption} onClick={() => updateCategory("Kanye West")}>
                    Kanye West
                    {currentCategory === "Kanye West" ? (
                        <RadioButtonCheckedIcon />
                    ) : (
                        <RadioButtonUncheckedIcon />
                    )}
                </div>
                <div
                    className={style.categoryOption}
                    onClick={() => updateCategory("Chuck Norris")}
                >
                    Chuck Norris
                    {currentCategory === "Chuck Norris" ? (
                        <RadioButtonCheckedIcon />
                    ) : (
                        <RadioButtonUncheckedIcon />
                    )}
                </div>
                <div className={style.categoryOption} onClick={() => updateCategory("Anime")}>
                    Anime ( ͡° ͜ʖ ͡°)
                    {currentCategory === "Anime" ? (
                        <RadioButtonCheckedIcon />
                    ) : (
                        <RadioButtonUncheckedIcon />
                    )}
                </div>
                <div className={style.categoryOption} onClick={() => updateCategory("Programming")}>
                    Programming
                    {currentCategory === "Programming" ? (
                        <RadioButtonCheckedIcon />
                    ) : (
                        <RadioButtonUncheckedIcon />
                    )}
                </div>
            </div>
            <div className={style.settingButtonDiv}>
                <Fab
                    className={style.settingButton}
                    color="primary"
                    size="medium"
                    onClick={() => cancelClicked()}
                >
                    <ArrowBackIcon />
                </Fab>
            </div>
        </div>
    );
}

SwitchCategory.propTypes = {
    cancelClicked: PropTypes.func.isRequired,
    updateCategory: PropTypes.func.isRequired,
    currentCategory: PropTypes.string.isRequired,
};

export default SwitchCategory;
