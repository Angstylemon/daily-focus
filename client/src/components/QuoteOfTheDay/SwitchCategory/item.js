import RadioButtonCheckedIcon from "@material-ui/icons/RadioButtonChecked";
import RadioButtonUncheckedIcon from "@material-ui/icons/RadioButtonUnchecked";
import style from "./style.module.scss";

export default function CategoryItem({ value, text, selected, clickHandler }) {
    return (
        <div
            id={`quoteCategoryValue-${value}`}
            className={selected ? style.selectedCategoryOption : style.categoryOption}
            onClick={() => clickHandler(value)}
        >
            {text}
            {selected ? <RadioButtonCheckedIcon /> : <RadioButtonUncheckedIcon />}
        </div>
    );
}
