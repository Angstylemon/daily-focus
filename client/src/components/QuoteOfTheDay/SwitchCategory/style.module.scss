.container {
    display: flex;
    flex-direction: column;
    width: 340px;
    height: fit-content;
    background-color: #ffffff;
    border-bottom-left-radius: 25px;
    border-bottom-right-radius: 25px;
    box-shadow: 0px 4px 16px 8px rgba(0, 0, 0, 0.1);
}

.listHolder {
    padding-top: 1vh;
    overflow-y: auto;
    height: 175px;
    display: flex;
    flex-direction: column;
    align-items: center;
}
.categoryOption {
    padding: 1vh 1vw;
    margin-bottom: 1vh;
    border: 1px solid grey;
    border-radius: 25px;
    height: 3.5em;
    display: flex;
    align-items: center;
    width: 75%;
}

.settingButton {
    background-color: #30a0f5;
    margin: 10px;
}

.settingButtonDiv {
    display: flex;
    justify-content: flex-end;
    align-items: flex-end;
}
