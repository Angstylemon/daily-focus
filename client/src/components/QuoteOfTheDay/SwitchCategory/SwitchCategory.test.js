import { jssPreset } from "@material-ui/styles";
import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";
import { shallow } from "enzyme";

import SwitchCategory from "../SwitchCategory";

let categoryState = "taylor-swift";
const updateCategory = (newCategory) => {
    categoryState = newCategory;
};
const quoteCategories = [
    {
        value: "chuck-norris",
        text: "Chuck Norris",
    },
    {
        value: "taylor-swift",
        text: "Taylor Swift",
    },
    {
        value: "kanye-west",
        text: "Kanye West",
    },
    {
        value: "programming",
        text: "Programming",
    },
    {
        value: "anime",
        text: "Anime ( ͡° ͜ʖ ͡°)",
    },
];

beforeEach(() => {
    categoryState = quoteCategories[0].value;
});

test("it renders correctly", () => {
    const shallowRenderer = new ShallowRenderer();
    const snapshotComponent = shallowRenderer.render(
        <SwitchCategory
            categories={quoteCategories}
            updateCategory={updateCategory}
            currentCategory={categoryState}
            cancelClicked={() => {}}
        />
    );
    expect(snapshotComponent).toMatchSnapshot();
});

test("When the setting button is pressed, the cancel function is called", () => {
    const mockFunction = jest.fn();
    const wrapper = shallow(
        <SwitchCategory
            categories={quoteCategories}
            updateCategory={updateCategory}
            currentCategory={categoryState}
            cancelClicked={mockFunction}
        />
    );
    wrapper.find(".settingButton").simulate("click");
    expect(mockFunction.mock.calls.length).toBe(1);
});
