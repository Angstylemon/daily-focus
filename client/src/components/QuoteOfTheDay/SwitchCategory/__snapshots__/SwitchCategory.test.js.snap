// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
<div>
  <p
    className="content"
  >
    Loren Ipsum
  </p>
  <p
    className="author"
  >
    - Dolor
  </p>
  <div
    className="settingButtonDiv"
  >
    <WithStyles(ForwardRef(Fab))
      className="settingButton"
      color="primary"
      onClick={[Function]}
      size="medium"
    >
      <Memo />
    </WithStyles(ForwardRef(Fab))>
  </div>
</div>
`;
